src/client/renderer/r_state.c

```diff
@@ -917,6 +985,16 @@ void R_UseInterpolation(const vec_t lerp) {
 }
 
 /**
+ * @brief Uploads shell offset value to the currently loaded program.
+ */
+void R_UseShellOffset(const vec_t offset) {
+
+	if (r_state.active_program->UseShellOffset) {
+		r_state.active_program->UseShellOffset(offset);
+	}
+}
+
```

Comment:
Feels a little gross to expose this here, when only one program will ever use this. Is there a way for the shell program's generic Use function to check the current entity and its flags, and modify the uniform accordingly?